File: VB/DXWebApplication1/Controllers/HomeController.vb
Imports System
Imports System.Web.Mvc
Imports DXWebApplication1.Models

Namespace DXWebApplication1.Controllers
	Public Class HomeController
		Inherits Controller

		Private list As New PersonsList()

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView(list.GetPersons())
		End Function

		<HttpPost, ValidateInput(False)>
		Public Function EditingAddNew(ByVal person As Person) As ActionResult
			If ModelState.IsValid Then
				list.AddPerson(person)
			End If
			Return PartialView("GridViewPartial", list.GetPersons())
		End Function

		<HttpPost, ValidateInput(False)>
		Public Function EditingUpdate(ByVal personInfo As Person) As ActionResult
			If ModelState.IsValid Then
				list.UpdatePerson(personInfo)
			End If
			Return PartialView("GridViewPartial", list.GetPersons())
		End Function

		Public Function EditingDelete(ByVal personId As Integer) As ActionResult
			list.DeletePerson(personId)
			Return PartialView("GridViewPartial", list.GetPersons())
		End Function
	End Class
End Namespace

File: CS/DXWebApplication1/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using DXWebApplication1.Models;

namespace DXWebApplication1.Controllers {
    public class HomeController : Controller {
        PersonsList list = new PersonsList();

        public ActionResult Index() {
            return View();
        }

        public ActionResult GridViewPartial() {
            return PartialView(list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingAddNew(Person person) {
            if(ModelState.IsValid)
                list.AddPerson(person);
            return PartialView("GridViewPartial", list.GetPersons());
        }

        [HttpPost, ValidateInput(false)]
        public ActionResult EditingUpdate(Person personInfo) {
            if(ModelState.IsValid)
                list.UpdatePerson(personInfo);
            return PartialView("GridViewPartial", list.GetPersons());
        }

        public ActionResult EditingDelete(int personId) {
            list.DeletePerson(personId);
            return PartialView("GridViewPartial", list.GetPersons());
        }
    }
}